File: SMA/SMA/Modelos/Usuario.cs
using System;
using System.Collections.Generic;

namespace SMA.Modelos
{
    public partial class Usuario
    {
        public Usuario()
        {
            InverseUsuarioCreacionNavigation = new HashSet<Usuario>();
            Reporte = new HashSet<Reporte>();
            RolUsuarioCreacionNavigation = new HashSet<Rol>();
            RolUsuarioEliminacionNavigation = new HashSet<Rol>();
            UsuarioContraseña = new HashSet<UsuarioContraseña>();
            UsuarioRol = new HashSet<UsuarioRol>();
        }

        public int Id { get; set; }
        public string Nombre { get; set; }
        public string ApellidoPaterno { get; set; }
        public string ApellidoMaterno { get; set; }
        public DateTime? FechaNacimiento { get; set; }
        public DateTime? FechaCreacion { get; set; }
        public int? UsuarioCreacion { get; set; }
        public DateTime? FechaEdicion { get; set; }
        public int? UsuarioEdicion { get; set; }
        public DateTime? FechaEliminacion { get; set; }
        public int? UsuarioEliminacion { get; set; }
        public string Correo { get; set; }
        public byte[] Contraseña { get; set; }
        public bool Activo { get; set; }
        public int? IdSeccion { get; set; }
        public short? IdRolApp { get; set; }
        public int? IdEstatusUsuario { get; set; }

        public virtual RolApp IdRolAppNavigation { get; set; }
        public virtual Usuario UsuarioCreacionNavigation { get; set; }
        public virtual ICollection<Usuario> InverseUsuarioCreacionNavigation { get; set; }
        public virtual ICollection<Reporte> Reporte { get; set; }
        public virtual ICollection<Rol> RolUsuarioCreacionNavigation { get; set; }
        public virtual ICollection<Rol> RolUsuarioEliminacionNavigation { get; set; }
        public virtual ICollection<UsuarioContraseña> UsuarioContraseña { get; set; }
        public virtual ICollection<UsuarioRol> UsuarioRol { get; set; }
    }
}

File: SMA/SMA.Web/Web/src/app/models/usuario.ts
import { Rol } from '../models/rol';

export class Usuario {
  public id: number;
  public nombre: string;
  public apellidoPaterno: string;
  public apellidoMaterno: string;
  public correo: string;
  public contraseña: string;
  public contrasena: string;
  public confirmarContrasena: string;
  public contraseñaCadena: string;
  public activo: boolean;
  public fechaNacimiento: Date;
  public rol: Rol[];
  public recuperar: string;

  constructor() {
    this.id = 0;
    this.nombre = '';
    this.apellidoPaterno = '';
    this.apellidoMaterno = '';
    this.correo = '';
    this.contrasena = '';
    this.confirmarContrasena = '';
  }
}
